src/state/application/gasUpdater.ts

```diff
@@ -0,0 +1,30 @@
+import { useEffect } from 'react';
+import { useGasPrice } from '../../hooks/useGasPrice';
+import { useActiveWeb3React } from 'hooks';
+
+import { updateGasPrice } from './actions';
+import { ChainId } from '@uniswap/sdk';
+import { useAppDispatch, useAppSelector } from 'state';
+
+export default function GasUpdater(): null {
+  const dispatch = useAppDispatch();
+
+  const { chainId } = useActiveWeb3React();
+
+  const block = useAppSelector((state) => {
+    return state.application.blockNumber[chainId ?? ChainId.MATIC];
+  });
+
+  const { fetchGasPrice, gasPrice, gasPriceLoading } = useGasPrice();
+
+  useEffect(() => {
```

Comment:
This should probably not update every block.  

20,000 active users might cause alot of strain on gas matic